File: LC350.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:

        res = []

        d = {}

        for i in nums1:
            if i in d:
                d[i] += 1
            else:
                d[i] = 1
        
        for i in nums2:
            if i in d and d[i] > 0:
                d[i] -= 1
                res.append(i)
        
        return res

File: LC350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        
        vector<int> res;
        map <int,int> m;

        for(int i = 0; i < nums1.size(); i++)
            m[nums1[i]]++;
        
        for(int i = 0; i < nums2.size(); i++){
            if(m[nums2[i]] > 0){
                m[nums2[i]]--;
                res.push_back(nums2[i]);
            }
        }

        return res;

        return res;
    }

};
